### ISSUE TITLE
Next-gen documentation infrastructure

### ISSUE THREAD

**Issue by jezdez** [conda/conda-docs#777]
To simplify documentation maintenance and better fit with the rest of our automation work we should:

- [ ] move documentation building with Sphinx into GitHub actions (running in conda/docs repo)
- [ ] fetch conda subprojects from separate repos similar to how RTD subprojects work
- [ ] standardize on one Sphinx theme (e.g. via #750)
- [ ] store results on S3 or GitHub pages
- [ ] make sure old links keep working
- [ ] set up a documentation maintenance team

**Comment by travishathaway** [comment]
As we go forward in this epic, we may still consider whether we store built documentation on S3 or GitHub pages (decide by considering [SymPy incident](https://www.asmeurer.com/blog/posts/the-sympy-hackerrank-dmca-incident/))

**Comment by jezdez** [comment]
Yeah, the idea is to use the same patterns applied in https://github.com/pyscript/pyscript/pull/262 and deploy directly to S3 and not GitHub Pages.

**Comment by emirkmo** [comment]
Can you please add documentation for contributors on how to use sphinx to create/view the docs? Thanks!

**Comment by jezdez** [comment]
Just added #818 as an item above, thank you for the suggestion!
